tests/network/test_reaction_network.py

```diff
@@ -823,38 +824,120 @@ def test_mols_w_cuttoff(self):
+            ([1, 652], [40, 53]),
+            ([4], [5, 6]),
+        ]
+        node_strings = []
 
-        RN_loaded = copy.deepcopy(RN_pr_solved)
+        for rxn in react_prod:
+            node_str = ReactionNetwork.generate_node_string(rxn[0], rxn[1])
+            node_strings.append(node_str)
+
+        self.assertListEqual(
+            node_strings, ["19+PR_32,673", "41,992", "1+PR_652,40+53", "4,5+6"]
+        )
+
+    def test_identify_concerted_rxns_via_intermediates(self):
 
+        v1 = loadfn(
+            os.path.join(test_dir, "identify_concerted_intermediate_list_v1.json")
+        )
         with open(
-            os.path.join(test_dir, "RN_unittest_reactions_list.json"), "rb"
-        ) as handle:
-            reactions = pickle.load(handle)
+            os.path.join(
+                test_dir, "identify_concerted_via_intermediate_unittest_RN.pkl"
+            ),
+            "rb",
+        ) as input:
+            RN_loaded = pickle.load(input)
+
+        v1_processed = []
+        for r_old in v1:
+            a = r_old[0]
+            b = r_old[1]
+            if None in b:
+                b.remove(None)
+            if len(a) == 2 and len(b) == 2:
+                inter = 0
+                inter = set(a).intersection(set(b))
+                for i in inter:
+                    a.remove(i)
+                    b.remove(i)
+            a.sort()
+            b.sort()
+            c = [a, b]
+            v1_processed.append(c)
+
+        v2_unique, v2_all = RN_loaded.identify_concerted_rxns_via_intermediates(
+            RN_loaded, single_elem_interm_ignore=[]
+        )
+        RN_loaded.add_concerted_rxns(RN_loaded, v2_unique)
+        v1_set = set(map(lambda x: repr(x), v1_processed))
+        v2_set = set(map(lambda x: repr(x), v2_unique))
+        inter = list(v1_set.intersection(v2_set))
+        v1_v2 = list(map(lambda y: literal_eval(y), v1_set - v2_set))
+        v2_v1 = list(map(lambda y: literal_eval(y), v2_set - v1_set))
+        dumpfn(v2_unique, "add_concerted_unittest_rxn_list.json")
+
+        missed_by_v2 = []
+        for r in v1_v2:
+            node_str = ReactionNetwork.generate_node_string(r[0], r[1])
+            if node_str not in RN_loaded.graph.nodes:
+                missed_by_v2.append(r)
+
+        self.assertEqual(len(missed_by_v2), 0)
+        self.assertEqual(len(v1_v2), 11)
+        self.assertEqual(len(v2_v1), 0)
+        self.assertEqual(len(inter), 29214)
+        self.assertEqual(len(v2_unique), 29214)
+
+        (
+            v2_unique_iter2,
+            v2_all_iter2,
+        ) = RN_loaded.identify_concerted_rxns_via_intermediates(
+            RN_loaded, single_elem_interm_ignore=[]
+        )
+
+        v2_set_iter2 = set(map(lambda x: repr(x), v2_unique_iter2))
+        inter_iter2 = list(v1_set.intersection(v2_set_iter2))
+        self.assertEqual(len(inter_iter2), 0)
+        self.assertEqual(len(v2_unique_iter2), 2100)
+
+    def test_add_concerted_rxns(self):
```

Comment:
I was hoping to see some logic here where we loaded a json/pkl that held the list of concerted reactions generated with the original concerted reaction v1 code, then removed from that list all e.g. A+B->A+C reactions and then checked the resulting list against the concerted reaction v2 code. Then in the future, once we have the option of narrowing concerteds according to reaction center(s), we'd be able to take that same list and further remove reactions that do not follow our new reaction center logic and then similarly test against the resulting list. It might feel somewhat roundabout, but it is a more rigorous test because we're ensuring that the change between the previous version and the new version is doing exactly what we expect (e.g. removing A+B->A+C now, or removing multi-RC reactions in the future) rather than us just ending up regenerating our reference list and testing against that each time a big change is made.

Reply:
I did upload reaction list json from the original method and removed A+B->A+C type reactions as well as all the repeats. Then I check every possibility between this processed list of original reactions and the new one generated by the current method. I check if there are any reactions missed by the current version, I check the intersection between the two sets as well as the two types of differences in the two sets.

Reply:
Where is the logic to remove the A+B->A+C reactions as well as the repeats? I was hoping to see that in the testing itself. If you've done that in an external script, then it makes it less transparent.

Reply:
Its in the test, v1_processed is the list created straight from the dumped reaction list from the original method.

Reply:
Ahhhh, of course. Perfect. Please just add some clarifying comments - that v1_processed was from the first version of the code, then at the point where you prune v1 down to something that matches v2, say how v2 is different from v1 and note the code that is pruning out those reactions we've decided to not have any longer